src/core/raster/qgsrasterinterface.h

```diff
@@ -227,16 +212,20 @@ class CORE_EXPORT QgsRasterInterface
     //! Clone itself, create deep copy
     virtual QgsRasterInterface *clone() const = 0 SIP_FACTORY;
 
-    //! Returns a bitmask containing the supported capabilities
-    virtual int capabilities() const
-    {
-      return QgsRasterInterface::NoCapabilities;
-    }
+    // TODO QGIS 4.0 -- rename to interfaceCapabilities, to avoid confusion with QgsRasterDataProvider::providerCapabilities
+    // (which inherits this class)
 
     /**
-     *  Returns the raster interface capabilities in friendly format.
+     * Returns the capabilities supported by the interface.
+     */
+    virtual Qgis::RasterInterfaceCapabilities capabilities() const;
```

Comment:
I'd also suggest for QGIS 4.0 to rename this to interfaceCapabilities() to make it clearer. I find the current providerCapabilities() + capabilities() pair of functions prone to confusion.

Reply:
Scroll up a bit, I've already added a TODO 😉